### PR TITLE
Improvements to DynamoDB binding

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
Ensure context propagation in PutItem and other minor changes

Spin-off from PR adding contexts to input bindings

### CODE REVIEW

File: bindings/aws/dynamodb/dynamodb.go [hunk 1]
```diff
@@ -95,13 +94,8 @@ func (d *DynamoDB) Invoke(ctx context.Context, req *bindings.InvokeRequest) (*bi
...
 
 func (d *DynamoDB) getDynamoDBMetadata(spec bindings.Metadata) (*dynamoDBMetadata, error) {
-	b, err := json.Marshal(spec.Properties)
-	if err != nil {
-		return nil, err
-	}
-
 	var meta dynamoDBMetadata
-	err = json.Unmarshal(b, &meta)
+	err := mapstructure.WeakDecode(spec.Properties, &meta)
```
Comment (berndverst) [inline bindings/aws/dynamodb/dynamodb.go:98]: Same as the comment on the other PR - why is this necessary?

Comment (ItalyPaleAle) [reply bindings/aws/dynamodb/dynamodb.go:98]: Not strictly necessary, but avoids marshalling to JSON and back. It's just more efficient (and the module was already imported in Dapr).

**Review by berndverst — APPROVED** [review]
